### PR TITLE
Add Login/Logout feature

### LINKED ISSUE #9: Logging in/ signing up with Google
- Permission to access e-mail and calendar to utilize google calendar API

https://developers.google.com/calendar/quickstart/python

### PR DESCRIPTION
**Pull request by Kcheung42**
This fixes #9

- Routes for Logging in
- Route for Logging Out
- Create session token and pass to client
- Manage session blacklisting

### CODE REVIEW

**Comment by gerardopar** [comment]
Sweet 💪🏼!!

File: server/.env [hunk 1]
```diff
@@ -4,3 +4,5 @@ FLASK_APP=project.__init__.py
 DATABASE_URL=postgres://postgres:password@postgres-db:5432/db_dev
 DATABASE_TEST_URL=postgres://postgres:password@postgres-db:5432/db_test
 DATABASE_SECRET=myprecious
+OAUTH_CLIENT_ID=798726661513-p69g4p5hn1cmura8mgm0sth9kqlnjoke.apps.googleusercontent.com
+OAUTH_CLIENT_SECRET=w_1s_mqk1bZ1G0iFnmMNefCG
```
Comment (brian-e-haley) [inline server/.env]: Should we be committing these remotely? Yes it is private but this is bad practice.

File: server/Pipfile [hunk 2]
```diff
@@ -12,6 +12,11 @@ psycopg2-binary = "*"
 flask-testing = "*"
 flask-restx = "*"
 python-dotenv = "*"
+google-auth = "*"
+requests = "*"
+oauthlib = "==3.0.1"
```
Comment (brian-e-haley) [inline server/Pipfile]: Is this needed?

File: server/.env (hunk 1 above)
Comment (gerardopar) [reply server/.env]: We could gitignore the env file

File: server/project/api/auth_handler.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                token, requests.Request(),
+                current_app.config['OAUTH_CLIENT_ID'])
+
+            if idinfo['iss'] not in [
+                    'accounts.google.com', 'https://accounts.google.com'
+            ]:
+                raise ValueError('Wrong issuer.')
+
+            # ID token is valid. Get the user's Google Account ID from the decoded token.
+            userid = idinfo['sub']
```
Comment (bonnieli) [inline server/project/api/auth_handler.py]: i suggest changing this var into `user_id` for better readability

File: server/.env (hunk 1 above)
Comment (Kcheung42) [reply server/.env]: we should not!

File: server/Pipfile (hunk 2 above)
Comment (Kcheung42) [reply server/Pipfile]: yes, except for oauthlib, will remove that

**Pull request merged**